src/byte_streams.clj

```diff
@@ -102,17 +102,24 @@
     :else
     (g/type (eval x))))
 
+(defn- extract-class [^Type src]
+  (when (and (instance? Class (.type src))
+             (not (.wrapper src)))
+    (if (= src (normalize-type-descriptor 'bytes))
+      'bytes
+      (.getName ^Class (.type src)))))
+
 (defmacro def-conversion
   "Defines a conversion from one type to another."
   [[src dst :as conversion] params & body]
   (let [^Type src (normalize-type-descriptor src)
-        dst (normalize-type-descriptor dst)]
+        dst (normalize-type-descriptor dst)
+        src-meta (if-let [c (extract-class src)]
```

Comment:
Would it make sense to have a fn like `(class-meta src)`

```clj
(defn class-meta [^Type src]
  (if-let [c (extract-class src)
     {:tag c}
     {}))
```

or some variant of that? Maybe `extract-class` could be inlined?